File: tests/uasm/sample/if_else_chain/IfElseChainTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class IfElseChainTest : UdonSharpBehaviour
{
    public void Start()
    {
        int x = 5;
        int result = 0;
        if (x > 10)
        {
            result = 1;
        }
        else if (x > 3)
        {
            result = 2;
        }
        else
        {
            result = 3;
        }
        Debug.Log(result);
    }
}

File: tests/uasm/sample/if_else_chain/IfElseChainTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import type { UdonInt } from "@ootr/udon-assembly-ts/stubs/UdonTypes";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class IfElseChainTest extends UdonSharpBehaviour {
  Start(): void {
    const x: UdonInt = 5n as UdonInt;
    let result: UdonInt = 0n as UdonInt;
    if (x > (10n as UdonInt)) {
      result = 1n as UdonInt;
    } else if (x > (3n as UdonInt)) {
      result = 2n as UdonInt;
    } else {
      result = 3n as UdonInt;
    }
    Debug.Log(result);
  }
}
